running-requests/prompt-blueprints.mdx

```diff
@@ -0,0 +1,80 @@
+---
+title: "Prompt Blueprints"
+icon: "layer-group"
+---
+
+Prompt Blueprints are a core part of PromptLayer's architecture that let you work with multiple LLM providers using one simple format. They hide the specific details of each provider, so you can switch between different LLMs without needing to change your code.
+
+## Why Use Prompt Blueprints?
+
+- **Model Agnostic**: Communicate with various LLM providers through a common interface.
+- **Code Stability**: Avoid code changes when updating or swapping LLM models.
+- **Flexibility**: Easily integrate new providers as they become available.
+- **Simplicity**: Focus on crafting prompts without worrying about underlying API differences.
+
+## Accessing LLM Responses with Prompt Blueprints
+
+Instead of accessing the raw LLM response via `response["raw_response"]`, it's recommended to use the standardized `response["prompt_blueprint"]`. This ensures consistency across different providers.
+
+### Example
+
+```python
+response = promptlayer_client.run(
+    prompt_name="ai-poet",
+    input_variables={'topic': 'food'},
+)
+
+print(response["prompt_blueprint"]["prompt_template"]["messages"][-1]["content"][0]["text"])
+```
+
+With this approach, you can update from one provider to another (e.g., OpenAI to Anthropic) without any code changes.
+
+## Understanding Placeholder Messages
+
+Placeholder Messages are a powerful feature that allows you to inject messages into a prompt template at runtime. By using the `placeholder` role, you can define placeholders within your prompt template that can be replaced with full messages when the prompt is executed.
+
+For more detailed information on Placeholder Messages, including how to create and use them, please refer to our dedicated [Placeholder Messages Documentation](/features/prompt-registry/placeholder-messages) page.
+
+### Running a Template with Placeholders
```

Comment:
How do I built a blueprint with functions? I think it's important that these docs are complete. Missing a lot as of now (but better than nothing obviously)

Reply:
You mean to include instructions on how to build a prompt template with function/tools on the UI and then show in code how to use those?